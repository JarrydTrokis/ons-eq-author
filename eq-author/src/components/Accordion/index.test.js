import React from "react";
import { shallow } from "enzyme";
import Accordion, { Button, Body } from "./";

describe("Accordion", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Accordion title="foo">Accordion panel</Accordion>);
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should open and close accordion", () => {
    expect(wrapper.find(Body).prop("open")).toBeTruthy(); //Open by default

    wrapper.find(Button).simulate("click");
    expect(wrapper.find(Body).prop("open")).toBeFalsy(); //Close

    wrapper.find(Button).simulate("click");
    expect(wrapper.find(Body).prop("open")).toBeTruthy(); //Open
  });
});
