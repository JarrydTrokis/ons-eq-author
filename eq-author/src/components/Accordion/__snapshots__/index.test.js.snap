// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render 1`] = `
<Accordion__Container>
  <Accordion__Header>
    <Accordion__Title>
      <Accordion__Button
        aria-controls="accordion-foo"
        aria-expanded={true}
        data-test="accordion-foo-button"
        onClick={[Function]}
        open={true}
      >
        foo
      </Accordion__Button>
    </Accordion__Title>
  </Accordion__Header>
  <Accordion__Body
    aria-hidden={false}
    data-test="accordion-foo-body"
    id="accordion-foo"
    open={true}
  >
    <Accordion__DisplayContent
      open={true}
    >
      Accordion panel
    </Accordion__DisplayContent>
  </Accordion__Body>
</Accordion__Container>
`;
